app/scripts/migrations/140.ts

```diff
@@ -0,0 +1,251 @@
+import {
+  Caip25CaveatType,
+  Caip25EndowmentPermissionName,
+  addPermittedEthChainId,
+} from '@metamask/multichain';
+import { PermissionNames } from '../controllers/permissions';
+
+type GenericPermissionControllerSubject =
+  PermissionControllerSubjects<PermissionConstraint>[string];
+
+export const version = 140;
+
+/**
+ * A map of the networks built into the extension at the time of this migration
+ * to their chain IDs.
+ *
+ * @see https://github.com/MetaMask/metamask-extension/blob/5b5c04a16fb7937a6e9d59b1debe4713978ef39d/shared/constants/network.ts#L535
+ */
+const BUILT_IN_NETWORKS: ReadonlyMap<string, Hex> = new Map([
+  ['sepolia', '0xaa36a7'],
+  ['mainnet', '0x1'],
+  ['linea-sepolia', '0xe705'],
+  ['linea-mainnet', '0xe708'],
+]);
+
+/**
+ * This migration adds the `endowment:permitted-chains` permission to all Snaps
+ * that have the `endowment:ethereum-provider` permission, and sets the selected
+ * chain ID for each Snap to the current selected network chain ID.
+ *
+ * This is necessary following the Amon Hen v2 changes to the network controller
+ * and other related controllers, as the `endowment:permitted-chains` permission
+ * is now required to use the Ethereum provider and switch networks.
+ *
+ * To simplify the use for Snaps, we automatically add the
+ * `endowment:permitted-chains` permission with the current selected network
+ * chain ID to all Snaps that have the `endowment:ethereum-provider` permission.
+ *
+ * @param originalVersionedData - Versioned MetaMask extension state, exactly
+ * what we persist to dist.
+ * @param originalVersionedData.meta - State metadata.
+ * @param originalVersionedData.meta.version - The current state version.
+ * @param originalVersionedData.data - The persisted MetaMask state, keyed by
+ * controller.
+ * @returns Updated versioned MetaMask extension state.
+ */
+export async function migrate(originalVersionedData: {
+  meta: { version: number };
+  data: Record<string, unknown>;
+}) {
+  const versionedData = cloneDeep(originalVersionedData);
+  versionedData.meta.version = version;
+  versionedData.data = transformState(versionedData.data);
+  return versionedData;
+}
+
+/**
+ * Transform the MetaMask extension state to add the
+ * `endowment:permitted-chains` permission to all Snaps that have the
+ * `endowment:ethereum-provider` permission.
+ *
+ * If the `PermissionController` or `SelectedNetworkController` state is not
+ * found or is not an object, the state is returned as-is.
+ *
+ * @param state - The MetaMask extension state.
+ * @returns The updated MetaMask extension state.
+ */
+function transformState(state: Record<string, unknown>) {
+  const {
+    NetworkController: networkControllerState,
+    PermissionController: permissionControllerState,
+    SelectedNetworkController: selectedNetworkControllerState,
+  } = state;
+  if (!networkControllerState || !isObject(networkControllerState)) {
+    log.warn(
```

Comment:
🤔 where do these logs go? I'm more familiar with the pattern of sending these logs to sentry via:
`global.sentry?.captureException(...)`. Like [here for example](https://github.com/MetaMask/metamask-extension/blob/f856a3b423a4316f14d2e5d7cd0297f76566f70a/app/scripts/migrations/139.ts#L168).

Reply:
Not sure. I copied this from another migration, but I see now that the Sentry pattern is used in others. I'll swap it out for that.